### PR TITLE
Make auto refresh interval customizable

### LINKED ISSUE #3103: Customizable autoRefreshInterval on Status Page
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

With the countdown now added in #2878 it would be great if we could customize the `autoRefreshInterval` (as seen on https://github.com/louislam/uptime-kuma/blob/master/src/pages/StatusPage.vue#L435).

For my usecase I would like the dashboard to refresh every 30 seconds e.g.

### ✔️ Solution

Add the ability to customize the `autoRefreshInterval` in the sidebar when editing a status page.

### ❓ Alternatives

A configuration variable would also be sufficient for my usecase

### 📝 Additional Context

I am willing to make a pull request if helpful, does not seem like it would be too difficult to implement

### PR DESCRIPTION
**Pull request by laupse**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

I needed a lower refresh interval on status page, so it could be refresh without full reload when using a browser extension. I was this issue #3103 so  i implemented it. Now we can customize refresh interval in the edit panel

Fixes #3103

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [x] My changes generates no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by cyril59310** [comment]
Only the translation file en.json must be modified to avoid conflicts with Weblate.

**Comment by laupse** [comment]
> Only the translation file en.json must be modified to avoid conflicts with Weblate.

Indeed. Os the failed check i have update the PR  accordingly

File: src/pages/StatusPage.vue [hunk 1]
```diff
@@ -34,6 +34,14 @@
                     </div>
                 </div>
 
+                <div class="my-3">
+                    <label for="auto-refresh-interval" class="form-label">{{ $t("Refresh Interval") }}</label>
+                    <input
```
Comment (CommanderStorm) [inline src/pages/StatusPage.vue:39]: I don't know if a help text is necessary or if a refresh interval is self-explanatory.

What are your thoughts on adding something like this?

> The status page will refresh every {0} seconds

**Review by CommanderStorm — COMMENTED** [review]
Overall very reasonable 👍🏻, I have left a nit about a helptext

File: src/pages/StatusPage.vue [hunk 2]
```diff
@@ -708,21 +715,21 @@ export default {
             this.$root.publicGroupList = res.data.publicGroupList;
 
             this.loading = false;
+
+            // Configure auto-refresh loop
+            this.updateHeartbeatList();
```
Comment (chakflying) [inline src/pages/StatusPage.vue:720]: This line (720) being moved up here means that the status page will only fetch the heartbeat data after the status page data is loaded, which will add a round-trip delay. I don't think it's necessary?

File: src/pages/StatusPage.vue (hunk 1 above)
Comment (laupse) [reply src/pages/StatusPage.vue:39]: I think it is pretty much self explanatory. But i can make this change for sure

File: src/pages/StatusPage.vue (hunk 2 above)
Comment (laupse) [reply src/pages/StatusPage.vue]: You are rigth. I move back this where it was

File: src/pages/StatusPage.vue (hunk 1 above)
Comment (laupse) [reply src/pages/StatusPage.vue]: Description added

File: src/lang/en.json [hunk 3]
```diff
@@ -884,4 +886,4 @@
     "GrafanaOncallUrl": "Grafana Oncall URL",
     "Browser Screenshot": "Browser Screenshot",
     "What is a Remote Browser?": "What is a Remote Browser?"
-}
+}
```
Comment (CommanderStorm) [inline src/lang/en.json:889]: change seems accidental

```suggestion
}

```

**Comment by CommanderStorm** [comment]
Thanks for the change to the status page

All changes in this PR are small and uncontroversial
⇒ merging with junior maintainer approval

**Pull request merged**